orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -251,13 +635,46 @@ public void sessionFactoryCreated(SessionFactoryImplementor sessionFactoryImplem
 
 	@Override
 	public GridType overrideType(Type type) {
-		log.info( "overrideType:" + type.getName() + ";" + type.getReturnedClass() );
+		log.debugf( "overrideType: %s ; ReturnedClass: %s", type.getName(), type.getReturnedClass() );
 		GridType gridType = null;
-		if ( type.getName().equals( "com.orientechnologies.orient.core.id.ORecordId" ) ) {
+
+		if ( type.getReturnedClass().equals( ORecordId.class ) ) {
 			gridType = ORecordIdGridType.INSTANCE;
 		}
+		else if ( type.getReturnedClass().equals( ORidBag.class ) ) {
+			gridType = ORidBagGridType.INSTANCE;
+		} // persist calendars as ISO8601 strings, including TZ info
+		/*
+		 * else if ( type == StandardBasicTypes.CALENDAR ) { gridType = Iso8601CalendarGridType.DATETIME_INSTANCE; }
+		 * else if ( type == StandardBasicTypes.CALENDAR_DATE ) { gridType = Iso8601CalendarGridType.DATE_INSTANCE; }
+		 * else if ( type == StandardBasicTypes.DATE ) { return Iso8601DateGridType.DATE_INSTANCE; } else if ( type ==
+		 * StandardBasicTypes.TIME ) { return Iso8601DateGridType.DATETIME_INSTANCE; } else if ( type ==
+		 * StandardBasicTypes.TIMESTAMP ) { return Iso8601DateGridType.DATETIME_INSTANCE; }
+		 */
+
+		// persist calendars as ISO8601 strings, including TZ info
+		else if ( type == StandardBasicTypes.CALENDAR ) {
+			return Iso8601StringCalendarType.DATE_TIME;
+		}
+		else if ( type == StandardBasicTypes.CALENDAR_DATE ) {
+			return Iso8601StringCalendarType.DATE;
+		}
+		// persist date as ISO8601 strings, in UTC, without TZ info
+		else if ( type == StandardBasicTypes.DATE ) {
+			return Iso8601StringDateType.DATE;
+		}
+		else if ( type == StandardBasicTypes.TIME ) {
+			return Iso8601StringDateType.TIME;
+		}
+		else if ( type == StandardBasicTypes.TIMESTAMP ) {
+			return Iso8601StringDateType.DATE_TIME;
+		}
+		else if ( type instanceof SerializableToBlobType ) {
+			SerializableToBlobType<?> exposedType = (SerializableToBlobType<?>) type;
+			return new SerializableAsStringType<>( exposedType.getJavaTypeDescriptor() );
```

Comment:
It seems you disabled the tests for Lobs but you are actually checking them. I think you can re-enable them.
Check my previous comment where I mention CouchDB